packages/webpack/src/utils/module-federation/share.ts

```diff
@@ -66,14 +76,59 @@ export function shareWorkspaceLibraries(
         (aliases, library) => ({ ...aliases, [library.name]: library.path }),
         {}
       ),
-    getLibraries: (eager?: boolean): Record<string, SharedLibraryConfig> =>
-      pathMappings.reduce(
-        (libraries, library) => ({
+    getLibraries: (eager?: boolean): Record<string, SharedLibraryConfig> => {
+      let pkgJson = null;
+      if (
+        project?.root &&
+        existsSync(
+          joinPathFragments(workspaceRoot, project.root, 'package.json')
+        )
+      ) {
+        pkgJson = readJsonFile(
+          joinPathFragments(workspaceRoot, project.root, 'package.json')
+        );
+      }
+      return pathMappings.reduce((libraries, library) => {
+        // Check to see if the library version ranged is declared in the root package.json
```

Comment:
`root package.json` sounds like the workspace's `package.json`, maybe just say `project's pacakge.json` or leave the comment off?